lib/validation.js

```diff
@@ -109,6 +109,39 @@ module.exports = function (yargs, usage) {
     }
   }
 
+  // validate arguments limited to enumerated choices
+  self.limitedChoices = function (argv) {
+    var options = yargs.getOptions()
+    var invalid = {}
+
+    if (!Object.keys(options.choices).length) return
+
+    Object.keys(argv).forEach(function (key) {
+      if (key !== '$0' && key !== '_' &&
+        options.choices.hasOwnProperty(key)) {
+        [].concat(argv[key]).forEach(function (value) {
+          // TODO case-insensitive configurability
```

Comment:
What do you think about having `TODO` comments in code? Should this be left to issues? I kinda forgot to pull it out.